### PR TITLE
feat: M3 withdrawals tests

### PR DESCRIPTION
**Pull request by johnnyonline**

### CODE REVIEW

File: test/integration/M3/Setup.t.sol [hunk 1]
```diff
@@ -0,0 +1 @@
+// TODO
```
Comment (danoctavian) [inline test/integration/M3/Setup.t.sol:1]: is there any benefit of this file?

File: test/integration/M3/Base.t.sol [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    IEigenPodManager public eigenPodManager;
+    IDelegationManager public delegationManager;
+
+    // Mock Contracts to deploy
+    BeaconChainMock public beaconChain;
+
+    // Ethereum
+    IDepositContract public depositContractEth2;
+
+    address public constant GLOBAL_ADMIN = 0x72fdBD51085bDa5eEEd3b55D1a46E2e92f0837a5;
```
Comment (danoctavian) [inline test/integration/M3/Base.t.sol:68]: this GLOBAL_ADMIN should NOT be used at all here.

It was a temporary measure for the preprod, we must not use it here in any way shape or form

File: test/integration/M3/Base.t.sol [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            beaconChain = new BeaconChainMock(EigenPodManager(address(eigenPodManager)), GENESIS_TIME_LOCAL);
+        }
+    }
+
+    function upgradeYnToM3() internal {
+        if (block.chainid != 17000) return;
+
+        // upgrade stakingNodesManager
+        {
+            vm.startPrank(ProxyAdmin(getTransparentUpgradeableProxyAdminAddress(address(stakingNodesManager))).owner());
```
Comment (danoctavian) [inline test/integration/M3/Base.t.sol:115]: let's not prank this

prank the PROXY_ADMIN

File: test/integration/M3/Base.t.sol [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+                ITransparentUpgradeableProxy(address(stakingNodesManager)),
+                address(new StakingNodesManager()),
+                ""
+            );
+            vm.stopPrank();
+        }
+
+        // upgrade ynETH
+        {
+            vm.startPrank(ProxyAdmin(getTransparentUpgradeableProxyAdminAddress(address(yneth))).owner());
```
Comment (danoctavian) [inline test/integration/M3/Base.t.sol:128]: let's not prank this

prank the PROXY_ADMIN_OWNER 

Same for all below.

File: test/integration/M3/Base.t.sol [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+            });
+            
+            vm.prank(GLOBAL_ADMIN);
+            stakingNodesManager.initializeV2(initParams);
+        }
+
+        // upgrade StakingNodeImplementation
+        {
+            stakingNodeImplementation = new StakingNode();
+            vm.prank(GLOBAL_ADMIN);
```
Comment (danoctavian) [inline test/integration/M3/Base.t.sol:201]: This should prank STAKING_ADMIN

File: test/integration/M3/Setup.t.sol (hunk 1 above)
Comment (johnnyonline) [reply test/integration/M3/Setup.t.sol]: https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/43c4269365f7cfdee928ac5abff1fce94cabd04a

File: test/integration/M3/Base.t.sol (hunk 2 above)
Comment (johnnyonline) [reply test/integration/M3/Base.t.sol]: https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/48f72390ec79e2ad439f89d86f244eb6ebc0c7e5

File: test/integration/M3/Base.t.sol (hunk 5 above)
Comment (johnnyonline) [reply test/integration/M3/Base.t.sol]: https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/48f72390ec79e2ad439f89d86f244eb6ebc0c7e5

File: test/integration/M3/Base.t.sol (hunk 4 above)
Comment (johnnyonline) [reply test/integration/M3/Base.t.sol]: https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/1069ccfd1cb7cedb873b964c87279e93206bd279

File: test/integration/M3/Base.t.sol (hunk 3 above)
Comment (johnnyonline) [reply test/integration/M3/Base.t.sol]: https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/1069ccfd1cb7cedb873b964c87279e93206bd279

**Review by danoctavian — APPROVED** [review]
LGTM

**Pull request merged**